R/pkg/R/functions.R

```diff
@@ -1369,6 +1369,39 @@ setMethod("quarter",
             column(jc)
           })
 
+#' @details
+#' \code{percentile_approx} Returns the approximate percentile value of
+#' numeric column at the given percentage.
+#'
+#' @param percentage numeric percentage at which percentile should be computed
+#'                   All values should be between 0 and 1.
+#'                   If length equals to 1 resulting column is of type double,
+#'                   otherwise array<double>/
+#' @param accuracy parameter (default: 10000) is a positive numeric literal which
+#' controls approximation accuracy at the cost of memory. Higher value of accuracy
+#' yields better accuracy, 1.0/accuracy is the relative error of the approximation.
+#'
+#' @rdname column_aggregate_functions
+#' @aliases percentile_approx percentile_approx,Column-method
+#' @note percentile_approx since 3.1.0
+setMethod("percentile_approx",
+          signature(x = "characterOrColumn", percentage = "numeric"),
```

Comment:
Can you make it as "numericOrColumn"`? I checked `ApproximatePercentile` and it seems allowing foldable expressions. It seems pretty much unlikely but still possible.